### DISCUSSION TITLE
Detection after several seconds

### DISCUSSION THREAD

**Discussion by cbroissa-github** [roleoroleo/yi-hack-Allwinner-v2#472] [Q&A]
Hello,
I have 4 cameras, 2 Yi 1080p Home IFUS and 2 Yi Outdoor 1080p IFUS.
I use the integration for Home Assistant with MQTT.
I use WEB RTC, the stream is real time, it's good (sometimes the stream doesn't work for a camera and work after a few minutes).

My problem is with MQTT and automation.
When I detect a motion, I want to take a photo for example, but the photo is taken after 10 seconds or more.

An idea of my problem ?

Thank you.

**Comment by Staini** [comment]
It's the message receiving/sending delay.
You can try get the local snapshots (saved on each cam), to reduce the delay... than it should be ~2 seconds.

**Comment by roleoroleo** [comment]
> Hello, I have 4 cameras, 2 Yi 1080p Home IFUS and 2 Yi Outdoor 1080p IFUS. I use the integration for Home Assistant with MQTT. I use WEB RTC, the stream is real time, it's good (sometimes the stream doesn't work for a camera and work after a few minutes).
> 
> My problem is with MQTT and automation. When I detect a motion, I want to take a photo for example, but the photo is taken after 10 seconds or more.
> 
> An idea of my problem ?
> 
> Thank you.

How do you take a photo? Are you using the mqtt "Topic Suffix for jpeg image"?
Is swap file enabled?
